internal/binder/function/funcs_str.go

```diff
@@ -357,4 +358,30 @@ func registerStrFunc() {
 		val:   ValidateOneStrArg,
 		check: returnNilIfHasAnyNil,
 	}
+	builtins["hex2dec"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			arg0 := strings.TrimPrefix(cast.ToStringAlways(args[0]), "0x")
+			dec, err := strconv.ParseInt(arg0, 16, 64)
+			if err != nil {
+				return fmt.Errorf("invalid hexadecimal value: %v", arg0), false
+			}
+			return dec, true
+		},
+		val:   ValidateOneStrArg,
+		check: returnNilIfHasAnyNil,
+	}
+	builtins["dec2hex"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			arg0, err := cast.ToInt(args[0], cast.STRICT)
+			if err != nil {
+				return err, false
+			}
+			hex := "0x" + strconv.FormatInt(int64(arg0), 16)
```

Comment:
hexStr := fmt.Sprintf("0x%X", num).  Maybe do a benchmark to check which one is faster.